### ISSUE TITLE
Default distance mode option, square or circular?

### ISSUE THREAD

**Issue by i2amroy** [CleverRaven/Cataclysm-DDA#13483]
So there's been some discussion about what the default distance mode should be, square or circular. It was cluttering up the original location, so I've made this issue for further discussion.

Personally I'm against switching the default to circular distances. Square distances are by far the more common in the two in rougelikes. Additionally while I think that weighted 8-dir movement works great in games with intelligent (or pseudo-intelligent) enemies in them, the fact that a lot of our monsters are mindless stumblers makes diagonal penalties hit them much harder than it would any sort of straight-line towards the target monster.

Now I'm not necessarily saying that I'd be totally deadlocked against switching to circular distances, but I think that if we want to do that then first we need to take a long, hard look and go over the exact stumbling mechanics with a fine tooth come to ensure that they influence overall speed as minimally as possible.

Having to double monster speed (probably the most powerful balancing tool) or halve move cost just to maintain the same balance on a monster with or without the STUMBLES flag is almost certainly a sign that that particular flag is having too much of an impact and steps should be taken to either lessen it's effects or split it into multiple levels, thus allowing for a more finely incremented balance change.
##

**Comment by kevingranade** [comment]
Uh, you made square vs circular it's own issue, but then you drug the whole
other issue into this one :/
The goal of the other PR is to neutralize the distinction re: cost of
stumbling, so it's irrelevant as long as we can achieve that goal.
I don't much care what other roguelikes do, especially since I don't
consider dda to be a roguelike.
The important thing to me is for the default to make intuitive sense, which
means trig distance, I.e. WYSIWYG. (Modulo non-square fonts, but a square
font should be the default anyway)

**Comment by illi-kun** [comment]
AFAIK nobody likes to see square blasts, craters, and spots of light in the game. Borders of craters are equidistant to epicenter of blast, and, generally, the form of all named effects depends on trig rules.
Player and mosters have to use the same logic as blast waves and particles of light (in macroworld) for keeping consistency - shortest distances should be defined identically in all cases.

Another argument for circular logic is people who coding the game don't need to think about alternative trig rules only in case of use IRL geometry (you need to replace your common sense by set of weird geometrical rules in other case, and that's obviously harder and may cause various logical errors).

**Comment by Rivet-the-Zombie** [comment]
illi-kun said it far better than I could.

I'm completely in favor of circular as default.

**Comment by Coolthulhu** [comment]
I'm against circular.

* Most of the mapgen is obviously working in squares, often with visible edges.
* Many effects still use a system similar to old explosions (for example, smokers, flashbangs) where the squares are obvious.
* Tactical movement is more dangerous in circular, because diagonally moving onto a tile with high move cost multiplies all of that.
* There is a discrepancy between reported `trig_dist` and actual distance at close range due to truncation to int. This requires extra special cases regardless of whether circular is only an option or the only option.

**Comment by kevingranade** [comment]
All your points are concerns with implementing trig dist, whether we
support it or not isn't on the table, the only question is whether to make
it the default.

> Most of the mapgen is obviously working in squares, often with visible
edges.

This is irrelevant, the structures we create are often square, but e.g.
craters *should* be circular, regardless of the distance calculation method.

> Many effects still use a system similar to old explosions (for example,
smokers, flashbangs) where the squares are obvious.

I'm not sure what you mean by, 'the squares are obvious'.  If the effect is
expanding in a square now, it should be switched to use an implementation
that invokes a option-aware primitive so it hits a square in square
distance and a circle in circular distance.

> Tactical movement is more dangerous in circular, because diagonally
moving onto a tile with high move cost multiplies all of that.

That's true, but since the enemies follow the same system, it should
generally balance out.

> There is a discrepancy between reported trig_dist and actual distance at
close range due to truncation to int. This requires extra special cases
regardless of whether circular is only an option or the only option.

This is just an implementation issue, I don't see how it influences what
should be the default.

If there are some bugs with trig distance, I'd buy a result of, 'we should
make trig distance the default, but only once these bugs are fixed'.

**Comment by i2amroy** [comment]
>Uh, you made square vs circular it's own issue, but then you drug the whole
other issue into this one :/

Reasons != implementations :P

Ideally discussion in the PR should be aimed at how we can implement things and fix things in that particular PR (such as how we could make speed approximately the same, etc.). This discussion, on the other hand, would just say "we need to fix this before I'd vote circular the default", without any too in-depth discussion of *how* we would fix it. 

Plus there's the simple fact that by stashing this in an issue we still have easy access to it and are reminded about it even if the PR gets merged. Even if we end up duplicating all the same discussion over here, it's still probably a better location for it than the monster speed PR (since not all of the bugs/problems we've pointed out involve monster speed, that's just one particularly gregarious example).

>If there are some bugs with trig distance, I'd buy a result of, 'we should make trig distance the default, but only once these bugs are fixed'.

I'd be fine with taking this approach.

**Comment by illi-kun** [comment]
### This issue was closed as it appears inactive.

Reducing open issues to those which are (or will) be actively worked upon helps us focus our efforts. This issue has not been deleted - it still appears in searches and if it contains relevant information you are encouraged to continue to link to it.

### If this issue was a bug
It should be reopened if it can be reproduced in the current build. You can obtain the most recent copy [here](http://dev.narc.ro/cataclysm/jenkins-latest/). Please check there is not a more recent report of this bug before doing so. If no more recent report exists you should continue the discussion in this issue.

### If this was a feature request
If the consensus was that the idea was good you could consider submitting an implementation via a PR. If you want to comment further please do so here as opposed to opening a new issue. Before posting check nobody has already made the same point and consider whether your comments are likely to lead to an implementation. If you have doubts about either consider instead [voting](http://iv.cataclysmdda.com:8111/all) for the issue

### If you want to work on this issue
Then either assign it to yourself or if you are unable to do so claim it via adding a comment. Please don't assign others or make a general request for action.
